docs/developers/releasing.md

```diff
@@ -0,0 +1,110 @@
+```
+
+### Commit release-notes(manually)
+After the release note is ready, commit it to `docs/CHANGELOG/CHANGELOG-{version}.md` of the minor release branch.
+
+### Prepare contributor-list(manually)
+Every release needs to indicate the contributors. Compare the newly created minor release branch with the previous minor release tag, for example, compare `release-1.4` branch with `v1.3.0` tag, to get contributors' Github ID list, the list should be in alphabetical order, like:
+```md
+## Contributors
+Thank you to everyone who contributed to this release!
+
+Users whose commits are in this release (alphabetically by username)
+@a
+@B
+@c
+@D
+...
+```
+
+### Update manifest(manually)
+When installing `Karmada`, the images needed to be pulled from DockerHub/SWR, so we should update manifests with image tags of the new version in the minor release branch. The following files need to be updated:
+* `charts/karmada/values.yaml`: Update `Karmada` related image tag with the new version.
+* `charts/index.yaml`: Add helm repository index.
+ 
+### Create release(manually)
+Now, all the preparations have been done, let's create a release on the release page.  
+* Create a new minor release tag, the tag name format should follow `v{version}`, for example, `v1.4.0`. 
+* The target branch is the newly created minor release branch.
+* The content of `Describe this release` should be the combination of the chapter `Prepare release-notes` and the chapter `Prepare contributor-list`.
+
+
+### Attach asserts(automatically)
+After the release is published, GitHub will run workflow `.github/workflows/release.yml` to build `karmadactl` and `kubectl-karmada` and attach them to the newly published release.
+
+
+### Build/Push images(automatically)
+After the release is published, GitHub will run `.github/workflows/swr-released-image.yml` and `.github/workflows/dockerhub-released-image.yml` to build all `Karmada` components' images and push them to DockerHub/SWR.
+
+### Verifying release(manually)
+After all the workflows have been finished, you should perform manual checks to see if the release came out correctly:
+ * Check if all required assets are attached.
+ * Check if all the required images have been published on DockerHub/SWR.
+
+## Patch release
+The patch version should be released from the corresponding minor release branch.
+
+### Prepare release-note(manually)
+This step is almost the same as the minor release, but we need to compare the minor release branch with the minor tag to extract the release note, for example, compare `release-1.3` branch with `v1.3.0` tag to collect `v1.3.1` patch release note.
+
+### Prepare contributor-list(manually)
+This step is almost the same as the minor release, but we need to compare the minor release branch with the minor tag to extract the contributors' Github ID list, for example, compare `release-1.3` branch with `v1.3.0` tag to collect `v1.3.1` patch release contributors' Github ID list.
```

Comment:
We don't list contributors for patch releases, Github will generate the list. See https://github.com/karmada-io/karmada/releases/tag/v1.4.1.